iroh-net/src/discovery/local_swarm_discovery.rs

```diff
@@ -71,126 +71,130 @@ impl LocalSwarmDiscovery {
+                                tokio::time::sleep(DISCOVERY_DURATION).await;
+                                trace!(?node_id, "discovery timeout");
+                                timeout_sender
+                                    .send(Message::Timeout(node_id, id))
+                                    .await
+                                    .ok();
+                            });
                         }
-                        let timeout_sender = task_sender.clone();
-                        timeouts.spawn(async move {
-                            tokio::time::sleep(DISCOVERY_DURATION).await;
-                            trace!(?node_id, "discovery timeout");
-                            timeout_sender
-                                .send(Message::Timeout(node_id, id))
-                                .await
-                                .ok();
-                        });
-                    }
-                    Message::Timeout(node_id, id) => {
-                        trace!(?node_id, "LocalSwarmDiscovery Message::Timeout");
-                        if let Some(senders_for_node_id) = senders.get_mut(&node_id) {
-                            senders_for_node_id.remove(&id);
-                            if senders_for_node_id.is_empty() {
-                                senders.remove(&node_id);
+                        Message::Timeout(node_id, id) => {
+                            trace!(?node_id, "LocalSwarmDiscovery Message::Timeout");
+                            if let Some(senders_for_node_id) = senders.get_mut(&node_id) {
+                                senders_for_node_id.remove(&id);
+                                if senders_for_node_id.is_empty() {
+                                    senders.remove(&node_id);
+                                }
                             }
                         }
-                    }
-                    Message::ChangeLocalAddrs(addrs) => {
-                        trace!(?addrs, "LocalSwarmDiscovery Message::ChangeLocalAddrs");
-                        discovery.remove_all();
-                        let addrs =
-                            LocalSwarmDiscovery::socketaddrs_to_addrs(addrs.direct_addresses);
-                        for addr in addrs {
-                            discovery.add(addr.0, addr.1)
+                        Message::ChangeLocalAddrs(addrs) => {
+                            trace!(?addrs, "LocalSwarmDiscovery Message::ChangeLocalAddrs");
+                            discovery.remove_all();
+                            let addrs =
+                                LocalSwarmDiscovery::socketaddrs_to_addrs(addrs.direct_addresses);
+                            for addr in addrs {
+                                discovery.add(addr.0, addr.1)
+                            }
                         }
                     }
                 }
             }
-        });
+            .instrument(info_span!("swarm-discovery.actor")),
```

Comment:
Only this `.instrument()` was added here.  All the other code is unchanged - despite what the diff may make you wonder.